cargo-shuttle/src/lib.rs

```diff
@@ -112,12 +112,21 @@ impl Shuttle {
         self.run(args, provided_path_to_init).await
     }
 
-    fn find_available_port(run_args: &mut RunArgs) {
+    fn find_available_port(run_args: &mut RunArgs, services_len: usize) {
         let mut available_port = 8010;
-        for port in (available_port..=std::u16::MAX).step_by(10) {
-            if portpicker::is_free_tcp(port) && portpicker::is_free_tcp(port + 1) {
-                available_port = port;
-                break;
+        for port in (available_port..=std::u16::MAX).step_by(services_len.max(10)) {
+            if portpicker::is_free_tcp(port) {
```

Comment:
This `port` is already checked in the inner loop, so we shouldn't need the if-clause